File: geophone-plus-2-python/REST/gppl2_response.py

from dataclasses import dataclass
from typing import Optional, List

@dataclass
class Provider:
    Name: Optional[str] = None
    City: Optional[str] = None
    State: Optional[str] = None
    LineType: Optional[str] = None
    Latitude: Optional[str] = None
    Longitude: Optional[str] = None

    def __str__(self) -> str:
        return (f"Provider: Name={self.Name}, City={self.City}, State={self.State}, "
                f"LineType={self.LineType}, Latitude={self.Latitude}, Longitude={self.Longitude}")

@dataclass
class Contacts:
    Name: Optional[str] = None
    Address: Optional[str] = None
    City: Optional[str] = None
    State: Optional[str] = None
    PostalCode: Optional[str] = None
    PhoneType: Optional[str] = None
    Latitude: Optional[str] = None
    Longitude: Optional[str] = None
    SICCode: Optional[str] = None
    SICDesc: Optional[str] = None
    QualityScore: Optional[str] = None

    def __str__(self) -> str:
        return (f"Contacts: Name={self.Name}, Address={self.Address}, City={self.City}, "
                f"State={self.State}, PostalCode={self.PostalCode}, PhoneType={self.PhoneType}, "
                f"Latitude={self.Latitude}, Longitude={self.Longitude}, SICCode={self.SICCode}, "
                f"SICDesc={self.SICDesc}, QualityScore={self.QualityScore}")

@dataclass
class PhoneInfo:
    Provider: Optional[Provider] = None
    Contacts: Optional[List[Contacts]] = None
    SMSAddress: Optional[str] = None
    MMSAddress: Optional[str] = None
    DateFirstSeen: Optional[str] = None
    DateOfPorting: Optional[str] = None
    NoteCodes: Optional[str] = None
    NoteDescriptions: Optional[str] = None
    TokensUsed: Optional[str] = None

    def __str__(self) -> str:
        contacts = ", ".join(str(c) for c in self.Contacts) if self.Contacts else "None"
        provider = str(self.Provider) if self.Provider else "None"
        return (f"PhoneInfo: Provider={provider}, Contacts=[{contacts}], SMSAddress={self.SMSAddress}, "
                f"MMSAddress={self.MMSAddress}, DateFirstSeen={self.DateFirstSeen}, "
                f"DateOfPorting={self.DateOfPorting}, NoteCodes={self.NoteCodes}, "
                f"NoteDescriptions={self.NoteDescriptions}, TokensUsed={self.TokensUsed}")

@dataclass
class Error:
    Type: Optional[str] = None
    TypeCode: Optional[str] = None
    Desc: Optional[str] = None
    DescCode: Optional[str] = None

    def __str__(self) -> str:
        return (f"Error: Type={self.Type}, TypeCode={self.TypeCode}, Desc={self.Desc}, "
                f"DescCode={self.DescCode}")

@dataclass
class GPPL2Response:
    PhoneNumber: Optional[str] = None
    TestType: Optional[str] = None
    LicenseKey: Optional[str] = None
    PhoneInfo: Optional[PhoneInfo] = None
    Error: Optional[Error] = None

    def __str__(self) -> str:
        phone_info = str(self.PhoneInfo) if self.PhoneInfo else "None"
        error = str(self.Error) if self.Error else "None"
        return (f"GPPL2Response: PhoneNumber={self.PhoneNumber}, TestType={self.TestType}, "
                f"LicenseKey={self.LicenseKey}, PhoneInfo={phone_info}")

File: geophone-plus-2-nodejs/src/SOAP/gppl2_response.js
export class Provider {
    constructor(data = {}) {
        this.Name = data.Name || null;
        this.City = data.City || null;
        this.State = data.State || null;
        this.LineType = data.LineType || null;
        this.Latitude = data.Latitude || null;
        this.Longitude = data.Longitude || null;
    }

    toString() {
        return `Provider: Name = ${this.Name}, City = ${this.City}, State = ${this.State}, LineType = ${this.LineType}, Latitude = ${this.Latitude}, Longitude = ${this.Longitude}`;
    }
}

export class Contacts {
    constructor(data = {}) {
        this.Name = data.Name || null;
        this.Address = data.Address || null;
        this.City = data.City || null;
        this.State = data.State || null;
        this.PostalCode = data.PostalCode || null;
        this.PhoneType = data.PhoneType || null;
        this.Latitude = data.Latitude || null;
        this.Longitude = data.Longitude || null;
        this.SICCode = data.SICCode || null;
        this.SICDesc = data.SICDesc || null;
        this.QualityScore = data.QualityScore || null;
    }

    toString() {
        return `Contacts: Name = ${this.Name}, Address = ${this.Address}, City = ${this.City}, State = ${this.State}, PostalCode = ${this.PostalCode}, PhoneType = ${this.PhoneType}, Latitude = ${this.Latitude}, Longitude = ${this.Longitude}, SICCode = ${this.SICCode}, SICDesc = ${this.SICDesc}, QualityScore = ${this.QualityScore}`;
    }
}

export class PhoneInfo {
    constructor(data = {}) {
        this.Provider = data.Provider ? new Provider(data.Provider) : null;
        this.Contacts = (data.Contacts || []).map(contact => new Contacts(contact));
        this.SMSAddress = data.SMSAddress || null;
        this.MMSAddress = data.MMSAddress || null;
        this.DateFirstSeen = data.DateFirstSeen || null;
        this.DateOfPorting = data.DateOfPorting || null;
        this.NoteCodes = data.NoteCodes || null;
        this.NoteDescriptions = data.NoteDescriptions || null;
        this.TokensUsed = data.TokensUsed || null;
    }

    toString() {
        const contactsString = this.Contacts.length
            ? this.Contacts.map(contact => contact.toString()).join(', ')
            : 'null';
        return `PhoneInfo: Provider = ${this.Provider ? this.Provider.toString() : 'null'}, Contacts = [${contactsString}], SMSAddress = ${this.SMSAddress}, MMSAddress = ${this.MMSAddress}, DateFirstSeen = ${this.DateFirstSeen}, DateOfPorting = ${this.DateOfPorting}, NoteCodes = ${this.NoteCodes}, NoteDescriptions = ${this.NoteDescriptions}, TokensUsed = ${this.TokensUsed}`;
    }
}

export class Error {
    constructor(data = {}) {
        this.Type = data.Type || null;
        this.TypeCode = data.TypeCode || null;
        this.Desc = data.Desc || null;
        this.DescCode = data.DescCode || null;
    }

    toString() {
        return `Error: Type = ${this.Type}, TypeCode = ${this.TypeCode}, Desc = ${this.Desc}, DescCode = ${this.DescCode}`;
    }
}

export class GPPL2Response {
    constructor(data = {}) {
        this.PhoneNumber = data.PhoneNumber || null;
        this.TestType = data.TestType || null;
        this.LicenseKey = data.LicenseKey || null;
        this.PhoneInfo = data.PhoneInfo ? new PhoneInfo(data.PhoneInfo) : null;
    }

    toString() {
        return `GPPL2Response: PhoneNumber = ${this.PhoneNumber}, TestType = ${this.TestType}, LicenseKey = ${this.LicenseKey}, PhoneInfo = ${this.PhoneInfo ? this.PhoneInfo.toString() : 'null'}`;
    }
}

export default GPPL2Response;